mteb/abstasks/AbsTaskInstructionRetrieval.py

```diff
@@ -0,0 +1,312 @@
+        top_ranked_ds = top_ranked_ds.cast_column('pid', Value('string'))
+        top_ranked_ds = top_ranked_ds.remove_columns([col for col in top_ranked_ds.column_names if col not in ['qid', 'pid']])
+        self.top_ranked = top_ranked_ds
+
+    
+    def load_corpus(self) -> Dict[str, Dict[str, str]]:
+        if not self.hf_repo:
+            self.check(fIn=self.corpus_file, ext="jsonl")
+
+        if not len(self.corpus):
+            logger.info("Loading Corpus...")
+            self._load_corpus()
+            logger.info("Loaded %d %s Documents.", len(self.corpus))
+            logger.info("Doc Example: %s", self.corpus[0])
+
+        return self.corpus
+    
+    def _load_corpus(self):
+        if self.hf_repo:
+            corpus_ds = load_dataset(self.hf_repo, 'corpus', keep_in_memory=self.keep_in_memory, streaming=self.streaming)
+        else:
+            corpus_ds = load_dataset('json', data_files=self.corpus_file, streaming=self.streaming, keep_in_memory=self.keep_in_memory)
+        corpus_ds = next(iter(corpus_ds.values())) # get first split
+        corpus_ds = corpus_ds.cast_column('_id', Value('string'))
+        corpus_ds = corpus_ds.rename_column('_id', 'id')
+        corpus_ds = corpus_ds.remove_columns([col for col in corpus_ds.column_names if col not in ['id', 'text', 'title']])
+        self.corpus = corpus_ds
+    
+    def _load_queries(self):
+        if self.hf_repo:
+            queries_ds = load_dataset(self.hf_repo, 'queries', keep_in_memory=self.keep_in_memory, streaming=self.streaming)
+        else:
+            queries_ds = load_dataset('json', data_files=self.query_file, streaming=self.streaming, keep_in_memory=self.keep_in_memory)
+        queries_ds = next(iter(queries_ds.values())) # get first split
+        queries_ds = queries_ds.cast_column('_id', Value('string'))
+        queries_ds = queries_ds.rename_column('_id', 'id')
+        queries_ds = queries_ds.remove_columns([col for col in queries_ds.column_names if col not in ['id', 'text', 'instruction_og', 'instruction_changed', "keywords", "short_query"]])
+        self.queries = queries_ds
+        
+    def _load_qrels(self, split, changed=False):
+        if self.hf_repo:
+            qrels_ds = load_dataset(self.hf_repo_qrels, "qrels_og" if not changed else "qrels_changed", keep_in_memory=self.keep_in_memory, streaming=self.streaming)[split]
+        else:
+            qrels_file = self.og_qrels_file if not changed else self.changed_qrels_file
+            qrels_ds = load_dataset('csv', data_files=qrels_file, delimiter='\t', keep_in_memory=self.keep_in_memory)
+        features = Features({'query-id': Value('string'), 'corpus-id': Value('string'), 'score': Value('float')})
+        qrels_ds = qrels_ds.cast(features)
+
+        if changed:
+            self.changed_qrels = qrels_ds
+        else:
+            self.og_qrels = qrels_ds
+
+
+class AbsTaskInstructionRetrieval(AbsTask):
```

Comment:
This task was needed as the evaluation requires evaluation twice over the queries, with two different instructions. 

It's modified to run with both sets of instructions and then compute the compared scores.

Reply:
Good ideas, thanks @KennethEnevoldsen!